pkg/puller/puller.go

```diff
@@ -244,23 +220,7 @@ func (p *Puller) recalcPeer(ctx context.Context, peer swarm.Address, po, d uint8
 		}
 		syncCtx.cancelBins(dontWant...)
 	} else {
-		// outside of depth
-		var (
-			want     = po
-			dontWant = []uint8{0} // never want bin 0
-		)
-
-		for i := uint8(0); i < p.bins; i++ {
-			if i == want {
-				continue
-			}
-			dontWant = append(dontWant, i)
-		}
-
-		if !syncCtx.isBinSyncing(want) {
-			p.syncPeerBin(ctx, syncCtx, peer, want, c[want])
-		}
-		syncCtx.cancelBins(dontWant...)
+		panic("shouldn't")
```

Comment:
Based on my comments above, I believe that out-of-neighborhood peers that are already syncing need to be cancelled.  That will cause this else to need to invoke syncCtx.cancelBins(dontWant...) where dontWant contains all possible bins.  This will shut down pullers that were started before the network depth was fully established.

Reply:
Something like:
	} else {
		// outside of depth, cancel them all
		var (
			dontWant = []uint8{0} // never want bin 0
		)

		for i := uint8(0); i < p.bins; i++ {
			dontWant = append(dontWant, i)
		}

		syncCtx.cancelBins(dontWant...)
	}